src/main/java/frc/robot/subsystems/intakeLimelight/IntakeLimelight.java

```diff
@@ -0,0 +1,579 @@
+     */
+    double scaling = diff_height / y_norm;
+    double x_target = scaling * x_norm;
+    double z_target = scaling * z_norm;
+    double distance = Math.hypot(z_target, x_target);
+
+    // Convert camera coordinates to our conventions
+    Translation2d result = new Translation2d(distance, new Rotation2d(z_target, -x_target));
+
+    // For NT so this function doesn't need to be called multiple times
+    m_lastDistance = distance;
+    m_lastX = result.getX();
+    m_lastY = result.getY();
+
+    return result;
+  }
+
+  /**
+   * Get a 2d translation from the camera to the target, including normalization to handle the
+   * effects of angle to target. See the below discussion.
+   * https://www.chiefdelphi.com/t/what-does-limelight-skew-actually-measure/381167/7
+   *
+   * @return Translation2d from the camera to the target. This is camera relative, x is out of the
+   *     camera and positive y is to the left from the camera's point of view.
+   */
+  public Translation2d getTargetTranslation() {
+    return getTargetTranslation(kTargetHeight);
+  }
+
+  /** Gets the object lowest in the image */
+  public Optional<Double> getXOfSmallestY(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    double minY = 10000000;
+    int minYIdx = 0;
+
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    for (int i = 1; i < corners.length; i += 2) {
+      if (corners[i] < minY) {
+        minY = corners[i];
+        minYIdx = i;
+      }
+    }
+
+    return Optional.of(corners[minYIdx - 1]);
+  }
+
+  public Optional<Double> getObjectHeightPx(double[] corners) {
+    // Format: x1 y1 x2 y2 x3 y3 . . .
+    if (corners.length == 0) {
+      return Optional.empty();
+    }
+
+    double minY = Double.MAX_VALUE;
+    double maxY = Double.MIN_VALUE;
+
+    for (int i = 1; i < corners.length; i += 2) {
+      minY = Math.min(minY, corners[i]);
+      maxY = Math.max(maxY, corners[i]);
+    }
+
+    return Optional.of(maxY - minY);
+  }
+
+  public class NoteDetection {
+    public double latencySec;
+
+    /** Distance from camera */
+    public double distanceMeters;
+
+    /** CCW Yaw Angle */
+    public double yawAngleDeg;
+
+    public NoteDetection(double latencySec, double distanceMeters, double yawAngleDeg) {
+      this.latencySec = latencySec;
+      this.distanceMeters = distanceMeters;
+      this.yawAngleDeg = yawAngleDeg;
+    }
+  }
+
+  /**
+   * Looks for a note
+   *
+   * @return If empty, no note detected. First value is note yaw in degrees (ccw), second value is
+   *     distance in meters.
+   */
+  public Optional<NoteDetection> getNotePos() {
+    // TODO filter low confidence detection in LL dashboard, hopefully
+    if (getPipeline() != pipeline.NOTE_PIPELINE) {
+      setPipeline(pipeline.NOTE_PIPELINE);
+    }
+
+    LimelightHelpers.LimelightResults llresults =
+        LimelightHelpers.getLatestResults(IntakeLimelightConstants.INTAKE_LIMELIGHT_NAME);
```

Comment:
There was something about the first time you run this, it takes a couple seconds? Something with the json parsing library. So you should make sure to do it before the match.

Not sure if doing it in the constructor works. Maybe disabledInit?

https://www.chiefdelphi.com/t/introducing-limelight-lib/425660/57?u=basel_a

Reply:
Added to `disabledInit`